core/src/display_object/avm2_button.rs

```diff
@@ -283,23 +283,37 @@ impl<'gc> Avm2Button<'gc> {
     pub fn set_state(self, context: &mut UpdateContext<'_, 'gc>, state: ButtonState) {
         self.invalidate_cached_bitmap(context.gc());
 
-        if let Some(state) = self.0.up_state.get() {
-            state.set_parent(context, None);
-        }
-        if let Some(state) = self.0.over_state.get() {
-            state.set_parent(context, None);
-        }
-        if let Some(state) = self.0.down_state.get() {
-            state.set_parent(context, None);
-        }
-        if let Some(state) = self.0.hit_area.get() {
+        for state in self.all_state_children(false) {
             state.set_parent(context, None);
         }
         if let Some(state) = self.get_state_child(state.into()) {
             state.set_parent(context, Some(self.into()));
         }
     }
 
+    /// Get all the display objects representing button states.
+    fn all_state_children(
+        &self,
+        weird_order: bool,
```

Comment:
I was going to say that this is not the most fortunate naming for a parameter, but:
 - I can't think of anything better...
 - I can see `weird_framescript_order` has been there before, and this is not worse...